src/disk_buffer_pool.cpp

```diff
@@ -67,17 +67,12 @@ namespace libtorrent
 	namespace {
 
 	// this is posted to the network thread
-	void watermark_callback(std::vector<boost::weak_ptr<disk_observer> >* cbs)
+	void watermark_callback(std::vector<boost::weak_ptr<disk_observer>> cbs)
```

Comment:
Good point. I've now confirmed with a simple test program that posting a std::bind and taking the parameter by const reference in the bound function does not produce any copies.